### PR TITLE
Components Showcase page revamp

### PR DESCRIPTION
**Pull request by masvelio**
## Link to Issue
Closes: #6107 

## Description of Changes
- Added completely new components page accessible from `/components` URL. 
- New page includes sidebar with scrollable content after clicking the item
- It is mobile friendly
- Added a `yarn add-component-showcase` script that helps engineers automate generation of new components in the proper structure/syntax etc. 

https://github.com/hicommonwealth/commonwealth/assets/14819225/6087490e-93b1-4031-a1e1-53a39c5de4c3

**Note: This PR includes scaffold and only few initial components. Created [another ticket](https://github.com/hicommonwealth/commonwealth/issues/6272) that will track the process of transitioning to the new components page.**

## Other Considerations

- Created ticket that will track the process of moving components to the new page https://github.com/hicommonwealth/commonwealth/issues/6272

### CODE REVIEW

**Review by Israellund — APPROVED** [review]
LGTM

File: knowledge_base/Component-Kit.md [hunk 1]
```diff
@@ -4,6 +4,22 @@ The component kit is Engineering’s current implementation of that source of tr
 
 A rendered version of our component kit can be [viewed on the live site](https://commonwealth.im/components), or found in the codebase in the `component_kit` directory.
 
+Components page got revamped and this also means that we will structure the components in a bit more readable way. There are two groups of building blocks in our kit:
```
Comment (gdjohnson) [inline knowledge_base/Component-Kit.md:7]: Possible to add a specific datestamp here? E.g. "As of 240109, the components page has been revamped"

File: knowledge_base/Component-Kit.md [hunk 2]
```diff
@@ -4,6 +4,22 @@ The component kit is Engineering’s current implementation of that source of tr
...
+2. Type a name for the new component. Make sure it is aligned with the design team. This name will be used as a file name in the codebase and a display name in component showcase page.
+3. Pick if new component should be of type "Foundations" or "Components". This should be aligned with the design team.
+4. New file is created in `/ComponentsShowcase/components` directory and `/ComponentsShowcase/componentsList.ts` file is updated.
+5. Edit newly created file to add content (display new component).
+
+### Engineering tips
+- In generated `{Component}.showcase.tsx` file use React Fragment `<> ... </>` as a wrapper for the whole component. 
+- Each generated file comes along with generated CSS class name that is identical as file name. If you want to add styles to specific showcase component, add proper class and styling in `ComponentsSwhocase.scss`.
+- There are two main util classes for handling the layout of this page (`flex-column` and `flex-row`). The idea is to minimize customization of styles in this page. Everything should be auto or semi generated.   
 ## Change Log
```
Comment (gdjohnson) [inline knowledge_base/Component-Kit.md:22]:
```suggestion

## Change Log
```

File: knowledge_base/Component-Kit.md [hunk 3]
```diff
@@ -4,6 +4,22 @@ The component kit is Engineering’s current implementation of that source of tr
 
 A rendered version of our component kit can be [viewed on the live site](https://commonwealth.im/components), or found in the codebase in the `component_kit` directory.
 
+Components page got revamped and this also means that we will structure the components in a bit more readable way. There are two groups of building blocks in our kit:
+- [Foundations](https://www.figma.com/file/yCUKk46hg9Fy3EXGkSoaKT/%F0%9F%9A%A7-Foundation?type=design&node-id=2%3A27&mode=design&t=qnBiTOOKaYhIM9bI-1) - the visual elements needed to create engaging end-to-end user experiences. This includes guidance on iconography, typography, layout and structure.
+- [Components](https://www.figma.com/file/eIVp33a1oCu0AtcLwSbGjr/%F0%9F%9A%A7-Components-and-Patterns?type=design&mode=design&t=LQy3p5zpdAM5fos8-1) - reusable building blocks of our design system. Each component meets a specific interaction or UI need, and has been specifically created to work together to create patterns and intuitive user experiences.
+
+### How to add component to components showcase page?
+1. Being in `/commonwealth` package, run `yarn add-component-showcase` command
```
Comment (gdjohnson) [inline knowledge_base/Component-Kit.md:12]:
```suggestion
1. From within the `/commonwealth` package, run the `yarn add-component-showcase` command
```

File: knowledge_base/Component-Kit.md [hunk 4]
```diff
@@ -4,6 +4,22 @@ The component kit is Engineering’s current implementation of that source of tr
...
+2. Type a name for the new component. Make sure it is aligned with the design team. This name will be used as a file name in the codebase and a display name in component showcase page.
+3. Pick if new component should be of type "Foundations" or "Components". This should be aligned with the design team.
+4. New file is created in `/ComponentsShowcase/components` directory and `/ComponentsShowcase/componentsList.ts` file is updated.
+5. Edit newly created file to add content (display new component).
+
+### Engineering tips
+- In generated `{Component}.showcase.tsx` file use React Fragment `<> ... </>` as a wrapper for the whole component. 
+- Each generated file comes along with generated CSS class name that is identical as file name. If you want to add styles to specific showcase component, add proper class and styling in `ComponentsSwhocase.scss`.
```
Comment (gdjohnson) [inline knowledge_base/Component-Kit.md:20]:
```suggestion
- Each generated file comes along with a generated CSS class name that is identical to the file name. If you want to add styles to specific showcase component, add proper class and styling in `ComponentsSwhocase.scss`.
```

File: knowledge_base/Component-Kit.md [hunk 5]
```diff
@@ -4,6 +4,22 @@ The component kit is Engineering’s current implementation of that source of tr
...
+- There are two main util classes for handling the layout of this page (`flex-column` and `flex-row`). The idea is to minimize customization of styles in this page. Everything should be auto or semi generated.   
 ## Change Log
 
+- 240109: Authored by Marcin Maslanka.
 - 230823: Authored by Graham Johnson.
```
Comment (gdjohnson) [inline knowledge_base/Component-Kit.md:25]:
```suggestion
```

Feel free to remove me in the change log—I basically just set up the page stub and probably shouldn't have been listed as author to begin with.

**Review by gdjohnson — COMMENTED** [review]
Thanks so much for adding documentation!

This new showcase page is beautiful. I remember the early days, it's come along way, and the autogeneration script seems super elegant. Haven't tested functionality yet, but have suggested some very minor docs changes.

Is it possible to add an entry for `add-component-showcase` in our `Package-Scripts.md` knowledge base file? The description can point to the `Component-Kit.md` file for more complete documentation, but it would be nice to just have a pointer reference there, since it's a central hub for all script documentation.

**Review by Miaplacidus — APPROVED** [review]
@masvelio 
Amazing! Love the look of the new components page. 👏
The yarn command works as expected. Thanks for adding!

File: knowledge_base/Component-Kit.md (hunk 1 above)
Comment (masvelio) [reply knowledge_base/Component-Kit.md]: done

File: knowledge_base/Component-Kit.md (hunk 2 above)
Comment (masvelio) [reply knowledge_base/Component-Kit.md:23]: done

File: knowledge_base/Component-Kit.md (hunk 3 above)
Comment (masvelio) [reply knowledge_base/Component-Kit.md]: done

File: knowledge_base/Component-Kit.md (hunk 5 above)
Comment (masvelio) [reply knowledge_base/Component-Kit.md]: done

**Comment by mzparacha** [comment]
Awesome work 🔥. The new page looks and feels amazing.

Just a very minor suggestion for follow-up tickets In https://github.com/hicommonwealth/commonwealth/issues/6272, "clicking on the color copies the code/variable to clipboard".

**Review by mzparacha — APPROVED** [review]

File: packages/commonwealth/scripts/add-component-showcase.ts [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+  const modifiedContent = contentWithImport
+    // Adds type to ComponentPageName
+    .replace(
+      /export const ComponentPageName = {([^}]+)}/,
+      `export const ComponentPageName = {$1  ${displayName}: '${displayName}',\n}`,
+    )
+    // Add object to componentItems array
+    .replace(
+      /export const componentItems = \[([^]+)\];/,
+      `export const componentItems = [$1  {\n    ComponentPage: ${displayName}Showcase,\n    displayName: ComponentPageName.${displayName},\n    type: ComponentType.${componentType},\n  },\n]`,
```
Comment (kurtisassad) [inline packages/commonwealth/scripts/add-component-showcase.ts:49]: So I think we need to add a semicolon at the end of the second argument of this replace string

Comment (masvelio) [reply packages/commonwealth/scripts/add-component-showcase.ts]: nice catch, your suggestion works!

**Pull request merged**